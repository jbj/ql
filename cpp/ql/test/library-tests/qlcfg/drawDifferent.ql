// query-type: graph
import Compare

from
  Element scopeElement,
  string scopeString, boolean isEdge, ControlFlowNode x, ControlFlowNode y, string label
where
  BothCFG::qltestGraph(scopeElement, scopeString, isEdge, x, y, label) and
  differentFunction(scopeElement)
select
  scopeString, isEdge, x, y, label
