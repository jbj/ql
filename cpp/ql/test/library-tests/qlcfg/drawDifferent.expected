| CPP-309__callNew_extractor | false | 88 | 88 | callNew |
| CPP-309__callNew_extractor | false | 96 | 96 | 5 |
| CPP-309__callNew_extractor | false | 99 | 99 | 2 |
| CPP-309__callNew_extractor | false | 102 | 102 | 3 |
| CPP-309__callNew_extractor | false | 103 | 103 | new[] |
| CPP-309__callNew_extractor | false | 105 | 105 | ExprStmt |
| CPP-309__callNew_extractor | false | 107 | 107 | return ... |
| CPP-309__callNew_extractor | false | 109 | 109 | { ... } |
| CPP-309__callNew_extractor | true | 99 | 102 |  |
| CPP-309__callNew_extractor | true | 102 | 103 |  |
| CPP-309__callNew_extractor | true | 103 | 107 |  |
| CPP-309__callNew_extractor | true | 105 | 99 |  |
| CPP-309__callNew_extractor | true | 107 | 88 |  |
| CPP-309__callNew_extractor | true | 109 | 105 |  |
| CPP-309__callNew_ql | false | 88 | 88 | callNew |
| CPP-309__callNew_ql | false | 96 | 96 | 5 |
| CPP-309__callNew_ql | false | 99 | 99 | 2 |
| CPP-309__callNew_ql | false | 102 | 102 | 3 |
| CPP-309__callNew_ql | false | 103 | 103 | new[] |
| CPP-309__callNew_ql | false | 105 | 105 | ExprStmt |
| CPP-309__callNew_ql | false | 107 | 107 | return ... |
| CPP-309__callNew_ql | false | 109 | 109 | { ... } |
| CPP-309__callNew_ql | true | 99 | 103 |  |
| CPP-309__callNew_ql | true | 103 | 107 |  |
| CPP-309__callNew_ql | true | 105 | 99 |  |
| CPP-309__callNew_ql | true | 107 | 88 |  |
| CPP-309__callNew_ql | true | 109 | 105 |  |
| CPP-309__callNew_unmodified | false | 88 | 88 | callNew |
| CPP-309__callNew_unmodified | false | 96 | 96 | 5 |
| CPP-309__callNew_unmodified | false | 99 | 99 | 2 |
| CPP-309__callNew_unmodified | false | 102 | 102 | 3 |
| CPP-309__callNew_unmodified | false | 103 | 103 | new[] |
| CPP-309__callNew_unmodified | false | 105 | 105 | ExprStmt |
| CPP-309__callNew_unmodified | false | 107 | 107 | return ... |
| CPP-309__callNew_unmodified | false | 109 | 109 | { ... } |
| CPP-309__callNew_unmodified | true | 99 | 102 |  |
| CPP-309__callNew_unmodified | true | 102 | 103 |  |
| CPP-309__callNew_unmodified | true | 103 | 107 |  |
| CPP-309__callNew_unmodified | true | 105 | 99 |  |
| CPP-309__callNew_unmodified | true | 107 | 88 |  |
| CPP-309__callNew_unmodified | true | 109 | 105 |  |
| stream_it__stream_it_extractor | false | 17688 | 17688 | stream_it |
| stream_it__stream_it_extractor | false | 17693 | 17693 | t |
| stream_it__stream_it_extractor | false | 17695 | 17695 | (reference dereference) |
| stream_it__stream_it_extractor | false | 17697 | 17697 | (reference to) |
| stream_it__stream_it_extractor | false | 17699 | 17699 | initializer for (__range) |
| stream_it__stream_it_extractor | false | 17704 | 17704 | declaration |
| stream_it__stream_it_extractor | false | 17706 | 17706 | declaration |
| stream_it__stream_it_extractor | false | 17707 | 17707 | declaration |
| stream_it__stream_it_extractor | false | 17708 | 17708 | { ... } |
| stream_it__stream_it_extractor | false | 17710 | 17710 | for(...:...) ... |
| stream_it__stream_it_extractor | false | 17712 | 17712 | return ... |
| stream_it__stream_it_extractor | false | 17714 | 17714 | { ... } |
| stream_it__stream_it_extractor | false | 17716 | 17716 | <error expr> |
| stream_it__stream_it_extractor | true | 17693 | 17699 |  |
| stream_it__stream_it_extractor | true | 17699 | 17716 |  |
| stream_it__stream_it_extractor | true | 17704 | 17693 |  |
| stream_it__stream_it_extractor | true | 17707 | 17708 |  |
| stream_it__stream_it_extractor | true | 17708 | 17716 |  |
| stream_it__stream_it_extractor | true | 17710 | 17704 |  |
| stream_it__stream_it_extractor | true | 17712 | 17688 |  |
| stream_it__stream_it_extractor | true | 17714 | 17710 |  |
| stream_it__stream_it_extractor | true | 17716 | 17707 | T |
| stream_it__stream_it_extractor | true | 17716 | 17712 | F |
| stream_it__stream_it_ql | false | 17688 | 17688 | stream_it |
| stream_it__stream_it_ql | false | 17693 | 17693 | t |
| stream_it__stream_it_ql | false | 17695 | 17695 | (reference dereference) |
| stream_it__stream_it_ql | false | 17697 | 17697 | (reference to) |
| stream_it__stream_it_ql | false | 17699 | 17699 | initializer for (__range) |
| stream_it__stream_it_ql | false | 17704 | 17704 | declaration |
| stream_it__stream_it_ql | false | 17706 | 17706 | declaration |
| stream_it__stream_it_ql | false | 17707 | 17707 | declaration |
| stream_it__stream_it_ql | false | 17708 | 17708 | { ... } |
| stream_it__stream_it_ql | false | 17710 | 17710 | for(...:...) ... |
| stream_it__stream_it_ql | false | 17712 | 17712 | return ... |
| stream_it__stream_it_ql | false | 17714 | 17714 | { ... } |
| stream_it__stream_it_ql | false | 17716 | 17716 | <error expr> |
| stream_it__stream_it_ql | true | 17693 | 17699 |  |
| stream_it__stream_it_ql | true | 17699 | 17706 |  |
| stream_it__stream_it_ql | true | 17704 | 17693 |  |
| stream_it__stream_it_ql | true | 17707 | 17708 |  |
| stream_it__stream_it_ql | true | 17710 | 17704 |  |
| stream_it__stream_it_ql | true | 17712 | 17688 |  |
| stream_it__stream_it_ql | true | 17714 | 17710 |  |
| stream_it__stream_it_unmodified | false | 17688 | 17688 | stream_it |
| stream_it__stream_it_unmodified | false | 17693 | 17693 | t |
| stream_it__stream_it_unmodified | false | 17695 | 17695 | (reference dereference) |
| stream_it__stream_it_unmodified | false | 17697 | 17697 | (reference to) |
| stream_it__stream_it_unmodified | false | 17699 | 17699 | initializer for (__range) |
| stream_it__stream_it_unmodified | false | 17704 | 17704 | declaration |
| stream_it__stream_it_unmodified | false | 17706 | 17706 | declaration |
| stream_it__stream_it_unmodified | false | 17707 | 17707 | declaration |
| stream_it__stream_it_unmodified | false | 17708 | 17708 | { ... } |
| stream_it__stream_it_unmodified | false | 17710 | 17710 | for(...:...) ... |
| stream_it__stream_it_unmodified | false | 17712 | 17712 | return ... |
| stream_it__stream_it_unmodified | false | 17714 | 17714 | { ... } |
| stream_it__stream_it_unmodified | false | 17716 | 17716 | <error expr> |
| stream_it__stream_it_unmodified | true | 17693 | 17699 |  |
| stream_it__stream_it_unmodified | true | 17699 | 17716 |  |
| stream_it__stream_it_unmodified | true | 17704 | 17693 |  |
| stream_it__stream_it_unmodified | true | 17707 | 17708 |  |
| stream_it__stream_it_unmodified | true | 17708 | 17716 |  |
| stream_it__stream_it_unmodified | true | 17710 | 17704 |  |
| stream_it__stream_it_unmodified | true | 17712 | 17688 |  |
| stream_it__stream_it_unmodified | true | 17714 | 17710 |  |
| stream_it__stream_it_unmodified | true | 17716 | 17707 | T |
| stream_it__stream_it_unmodified | true | 17716 | 17712 | F |
